src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -537,75 +554,78 @@ public Task<Transaction> MergeTransactionSignaturesAsync(uint256 depositId, Tran
             {
                 lock (this.lockObj)
                 {
-                    this.Synchronize();
-
-                    this.logger.LogInformation("ValidateCrossChainTransfers : {0}", depositId);
-                    ICrossChainTransfer transfer = this.ValidateCrossChainTransfers(this.Get(new[] { depositId })).FirstOrDefault();
-
-                    if (transfer == null)
+                    lock (((FederationWalletManager)this.federationWalletManager).lockObject)
```

Comment:
I see this pattern everywhere... 2 locks.

Reply:
We don't wallet manager to make any changes after we have synchronized ourselves with the wallet.